### PR TITLE
Regenerate parser for the observe mini-language

### LINKED ISSUE #1739: `sre_parse` and `sre_constants` modules are deprecated
`traits` makes use of the `sre_parse` and `sre_constants` modules [here](https://github.com/enthought/traits/blob/6ebef9089467495cdb8ab24c4991374be1d582e2/traits/observation/_generated_parser.py#L258-L264); however, these modules are deprecated, and a deprecation warning is emitted when they are imported.  They should be replaced with something more up-to-date.

### PR DESCRIPTION
**Pull request by mdickinson**
This PR regenerates the parser for the `observe` mini-language using the newest version of Lark. This should fix some deprecation warnings coming from imports used by the old parser.

The parser was regenerated using Lark 1.1.5, via

    python -m lark.tools.standalone traits/observation/_dsl_grammar.lark --out traits/observation/_generated_parser.py

Closes #1739

### CODE REVIEW

**Comment by mdickinson** [comment]
Tests are currently failing for unrelated reasons; we need #1741 to be merged.

File: traits/observation/_generated_parser.py [hunk 1]
```diff
@@ -1665,443 +2332,834 @@ def from_ParseTable(cls, parse_table):
...
...1, {'@': 31})}, 31: {17: (0, 24), 18: (0, 3), 8: (0, 29), 9: (0, 13)}, 32: {0: (0, 17), 1: (0, 23), 2: (0, 25), 3: (0, 18), 13: (0, 10), 4: (0, 14), 5: (0, 2), 6: (0, 4), 7: (0, 15)}, 33: {10: (1, {'@': 32}), 11: (1, {'@': 32})}, 34: {10: (1, {'@': 28}), 11: (1, {'@': 28})}, 35: {8: (1, {'@': 20}), 9: (1, {'@': 20}), 10: (1, {'@': 25}), 11: (1, {'@': 25})}}, 'start_states': {'start': 16}, 'end_states': {'start': 21}}, '__type__': 'ParsingFrontend'}, 'rules': [{'@': 10}, {'@': 11}, {'@': 12}, {'@': 13}, {'@': 14}, {'@': 15}, {'@': 16}, {'@': 17}, {'@': 18}, {'@': 19}, {'@': 20}, {'@': 21}, {'@': 22}, {'@': 23}, {'@': 24}, {'@': 25}, {'@': 26}, {'@': 27}, {'@': 28}, {'@': 29}, {'@': 30}, {'@': 31}, {'@': 32}, {'@': 33}], 'options': {'debug': False, 'keep_all_tokens': False, 'tree_class': None, 'cache': False, 'postlex': None, 'parser': 'lalr', 'lexer': 'contextual', 'transformer': None, 'start': ['start'], 'priority': 'normal', 'ambiguity': 'auto', 'regex': False, 'propagate_positions': False, 'lexer_callbacks': {}, 'maybe_placeholders': False, 'edit_terminals': None, 'g_regex_flags': 0, 'use_bytes': False, 'import_paths': [], 'source_path': None, '_plugins': {}}, '__type__': 'Lark'}
```
Comment (rahulporuri) [inline traits/observation/_generated_parser.py:3157]: I naively assumed that I could run `python -m lark.tools.standalone traits/observation/_dsl_grammar.lark --out traits/observation/_generated_parser.py` on my personal machine and check that there are no changes in the generated parser code.

That doesn't seem to be the case. This dictionary gets modified when I regenerate the parser code. The change just moved around the key/value pairs in the `parser` `tokens` dictionary.

Comment (mdickinson) [reply traits/observation/_generated_parser.py:3157]: Hmm, interesting; that'll be due to hash randomization, I guess. https://github.com/lark-parser/lark/issues/595 looks related.

Possibly we should be generating using a fixed `PYTHONHASHSEED`.

Comment (mdickinson) [reply traits/observation/_generated_parser.py:3157]: > It's not clear to me what the source of the non-determinism is.

From a brief look at the Lark source, I suspect that the source of non-determinism is memory addresses of objects: the code is using `id`s of objects as set elements and/or dictionary keys. (Which then affects iteration order, etc.) There _is_ a use of `random.randint` in `earley_forest.py`, but I don't think we exercise that code path.

I'll stop trying to find a way to make this deterministic and merge as-is.

**Pull request merged**